bcda/main.go

```diff
@@ -422,10 +422,11 @@ func createAlphaToken(ttl int, acoSize string) (string, error) {
 	if err != nil {
 		return "", fmt.Errorf("could not register client for %s (%s) because %s", aco.UUID.String(), aco.Name, err.Error())
 	}
-	aco.ClientID = creds.ClientID
+
 	db := database.GetGORMDbConnection()
 	defer database.Close(db)
-	err = db.Save(&aco).Error
+	// Only update ClientID; the AlphaSecret may already be in the database, and we'd overwrite it with an empty string
```

Comment:
This comment at first confused me, so I had to go read the code. The source of confusion is that the API and Alpha Auth are using the same table in the same database. Alpha auth needs to remember both the client_id and alpha_secret when it registers a client, but the API only remembers the client_id when it receives and forwards the credentials to output. This CLI command "belongs" to the API, so it only touches client_id. Finally, because this same command is used whether it's being called with Alpha Auth or Okta Auth, it has to save the client_id.

Reply:
Comment updated to something hopefully more useful.